--- origsrc/krb5-auth-dialog-0.16/plugins/Makefile.am	2010-04-27 12:51:14.000000000 -0500
+++ src/krb5-auth-dialog-0.16/plugins/Makefile.am	2010-06-30 17:07:24.380984000 -0500
@@ -27,6 +27,7 @@ libka_plugin_dummy_la_CPPFLAGS = \
 libka_plugin_dummy_la_LDFLAGS = \
 	-module \
 	-avoid-version \
+	-no-undefined -Wl,$(top_builddir)/src/libka-plugin.a \
 	$(NULL)
 
 # PAM plugin
@@ -44,6 +45,7 @@ libka_plugin_pam_la_CPPFLAGS = \
 libka_plugin_pam_la_LDFLAGS = \
 	-module \
 	-avoid-version \
+	-no-undefined -Wl,$(top_builddir)/src/libka-plugin.a \
 	$(PAM_LIBS) \
 	$(NULL)
 endif
--- origsrc/krb5-auth-dialog-0.16/src/Makefile.am	2010-04-29 03:28:46.000000000 -0500
+++ src/krb5-auth-dialog-0.16/src/Makefile.am	2010-06-30 17:34:42.244664500 -0500
@@ -70,6 +70,8 @@ krb5_auth_dialog_CFLAGS = \
 	$(AM_CFLAGS) \
 	$(NULL)
 
+krb5_auth_dialog_LDFLAGS = -export-dynamic -Wl,--out-implib,libka-plugin.a
+
 krb5_auth_dialog_LDADD = \
 	$(top_builddir)/gtksecentry/libgtksecentry.a \
 	$(top_builddir)/secmem/libsecmem.a \
--- origsrc/krb5-auth-dialog-0.16/src/ka-plugin-loader.c	2010-04-29 03:28:46.000000000 -0500
+++ src/krb5-auth-dialog-0.16/src/ka-plugin-loader.c	2010-06-30 17:10:34.547860900 -0500
@@ -120,7 +120,7 @@ load_plugins (KaPluginLoader *self)
 		char *fname;
 		KaPlugin *plugin;
 
-		fname = g_strdup_printf("libka-plugin-%s.%s", pname, G_MODULE_SUFFIX);
+		fname = g_strdup_printf("ka-plugin-%s", pname);
 		path = g_module_build_path (KA_PLUGINS_DIR, fname);
 
 		plugin = load_plugin (path);
