--- origsrc/krb5-auth-dialog-0.17/plugins/Makefile.am	2010-09-21 23:03:40.216909100 -0500
+++ src/krb5-auth-dialog-0.17/plugins/Makefile.am	2010-09-21 23:01:33.345417000 -0500
@@ -28,6 +28,7 @@ libka_plugin_dummy_la_CPPFLAGS = \
 libka_plugin_dummy_la_LDFLAGS = \
 	-module \
 	-avoid-version \
+	-no-undefined -Wl,$(top_builddir)/src/libka-plugin.a -lgobject-2.0 \
 	$(NULL)
 
 # PAM plugin
@@ -45,6 +46,7 @@ libka_plugin_pam_la_CPPFLAGS = \
 libka_plugin_pam_la_LDFLAGS = \
 	-module \
 	-avoid-version \
+	-no-undefined -Wl,$(top_builddir)/src/libka-plugin.a -lgobject-2.0 \
 	$(PAM_LIBS) \
 	$(NULL)
 endif
@@ -63,4 +65,5 @@ libka_plugin_afs_la_CPPFLAGS = \
 libka_plugin_afs_la_LDFLAGS = \
 	-module \
 	-avoid-version \
+	-no-undefined -Wl,$(top_builddir)/src/libka-plugin.a -lgobject-2.0 \
 	$(NULL)
--- origsrc/krb5-auth-dialog-0.17/src/Makefile.am	2010-09-21 23:03:40.232509100 -0500
+++ src/krb5-auth-dialog-0.17/src/Makefile.am	2010-09-21 22:24:49.000000000 -0500
@@ -70,6 +70,8 @@ krb5_auth_dialog_CFLAGS = \
 	$(AM_CFLAGS) \
 	$(NULL)
 
+krb5_auth_dialog_LDFLAGS = -export-dynamic -Wl,--out-implib,libka-plugin.a
+
 krb5_auth_dialog_LDADD = \
 	$(top_builddir)/gtksecentry/libgtksecentry.a \
 	$(top_builddir)/secmem/libsecmem.a \
--- origsrc/krb5-auth-dialog-0.17/src/ka-plugin-loader.c	2010-09-21 23:03:40.232509100 -0500
+++ src/krb5-auth-dialog-0.17/src/ka-plugin-loader.c	2010-09-21 22:24:49.000000000 -0500
@@ -120,7 +120,7 @@ load_plugins (KaPluginLoader *self)
 		char *fname;
 		KaPlugin *plugin;
 
-		fname = g_strdup_printf("libka-plugin-%s.%s", pname, G_MODULE_SUFFIX);
+		fname = g_strdup_printf("ka-plugin-%s", pname);
 		path = g_module_build_path (KA_PLUGINS_DIR, fname);
 
 		plugin = load_plugin (path);
